edgelet/iotedged/src/lib.rs

```diff
@@ -1459,4 +1521,89 @@ mod tests {
             proxy_val
         );
     }
+
+    #[test]
+    fn diff_with_same_cached_returns_false() {
+        let tmp_dir = TempDir::new("blah").unwrap();
+        let path = tmp_dir.path().join("cache");
+        let settings = Settings::new(Some(Path::new(GOOD_SETTINGS))).unwrap();
+        let settings_to_write = serde_json::to_string(&settings).unwrap();
+        let sha_to_write = Sha256::digest_str(&settings_to_write);
+        let base64_to_write = base64::encode(&sha_to_write);
+        File::create(&path)
+            .unwrap()
+            .write_all(base64_to_write.as_bytes())
+            .unwrap();
+        assert_eq!(diff_with_cached(&settings, &path), false);
```

Comment:
>assert_eq [](start = 8, length = 9)

This might be a style thing, but would `assert!(!diff_with_cached(&settings, &path))`  look better?

Reply:
Agree. It would.